src/merkle/store/mod.rs

```diff
@@ -440,6 +439,14 @@ impl From<&Mmr> for MerkleStore {
     }
 }
 
+impl From<&TieredSmt> for MerkleStore {
+    fn from(value: &TieredSmt) -> Self {
+        let mut store = MerkleStore::new();
+        store.extend(value.inner_nodes());
+        store
+    }
+}
```

Comment:
optional nit: this looks like a `FromIterator<Item = InnerNodeInfo>`. However, we might want to preserve explicit types here as creating a `MerkleStore` from a `&TieredSmt` is self-explanatory, as opposed to creating a `MerkleStore` from an iterator of `InnerNodeInfo` that happens to be created from a `TieredSmt`.